pkg/detectors/openai/openai.go

```diff
@@ -103,6 +70,67 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 	return
 }
 
+func verifyToken(ctx context.Context, client *http.Client, token string) (bool, map[string]string, error) {
+	// Undocumented API
+	// https://api.openai.com/v1/organizations
+	req, err := http.NewRequestWithContext(ctx, "GET", "https://api.openai.com/v1/organizations", nil)
+	if err != nil {
+		return false, nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	res, err := client.Do(req)
+	if err != nil {
+		return false, nil, err
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, res.Body)
+		_ = res.Body.Close()
+	}()
+
+	switch res.StatusCode {
+	case 200:
+		var orgs orgResponse
+		if err = json.NewDecoder(res.Body).Decode(&orgs); err != nil {
+			return false, nil, err
+		}
+
+		org := orgs.Data[0]
+		extraData := map[string]string{
+			"id":          org.ID,
+			"title":       org.Title,
+			"user":        org.User,
+			"description": org.Description,
+			"role":        org.Role,
+			"is_personal": strconv.FormatBool(org.Personal),
+			"is_default":  strconv.FormatBool(org.Default),
+			"total_orgs":  fmt.Sprintf("%d", len(orgs.Data)),
+		}
+		return true, extraData, nil
```

Comment:
@rgmz Metadata here is an additional information and the credentials are already been verified based on HTTP Status code. Base on the conversation we had on [#2807 ](https://github.com/trufflesecurity/trufflehog/pull/2807#discussion_r1594184573) and [#2808 ](https://github.com/trufflesecurity/trufflehog/pull/2808#discussion_r1596810738), these are non-fatal errors and can be ignored without affecting the verification result.

Reply:
> ..these are non-fatal errors and can be ignored without affecting the verification result.

The question I posed was actually whether it makes sense to return "verified" alongside an error to indicate a potential problem with the response. Generally speaking, I think it's much safer to handle errors rather than ignore them `if err == nil { ... }`.

There's a larger problem of surfacing errors and giving them precedence, but that's a different topic. _e.g.,_ the GitHub detector failing due to a DNS timeout and the JDBC detector failing because of an invalid hosts shouldn't be treated equally.

Reply:
You are right that these errors should not be ignored. Unfortunately, right now, Detector result struct does not supports to hold these kind of errors (other than verification). In future, we can have separate error field(s) like @dustin-decker proposed to name it as EnrichmentError.

Reply:
> Unfortunately, right now, Detector result struct does not supports to hold these kind of errors (other than verification).

I would still consider them verification errors. e.g., invalid JSON can be indicative of false-positives or the API not behaving as expected (#2099, #2218).

Reply:
Thanks @rgmz for sharing this case study. It is valid use case to convince me. One more case came in my mind is if an API started to redirected to valid page (with status 200) with HTML in response (may be home page), It will also be indicative of false-positives.

Reply:
Thanks @rgmz for sharing this case study. It is valid use case to convince me. One more case came in my mind is if an API started to redirected to valid page (with status 200) with HTML in response (may be home page), It will also be indicative of false-positives.